### PR TITLE
[New Rule] Disable Windows Event and Security Logs

### LINKED ISSUE #1071: [New Rule] Windows EventLog Disabled
## Description

Powershell command that sets the Windows EventLog to disabled.

### Target indexes

logs-windows-*, winlogbeat-*
### Target Operating Systems

windows
### Tested ECS Version
x.x.x
1.8.0

## Optional Info
Tactic: Defense Evasion(TA0005) Technique: Impair Defenses(T1562) Sub-Technique: T1562.002(disable windows event logging)
### Query
process where event.module == "powershell" and
process.args : ( "powershell.exe", "Set-Service", "EventLog", "Disabled")
## Example Data

<details>

```json
{
  "_id": "89933b5f64737a55c666fd1a7155b02c533e65e040dddfb611d83e563afa6796",
  "_index": ".siem-signals-siemplify-000002",
  "_score": "1",
  "_type": "_doc",
  "@timestamp": "2021-03-31T04:43:10.154Z",
  "agent": {
    "ephemeral_id": "af1d7136-94f2-42bd-8ac6-c0c320381735",
    "hostname": "DESKTOP-AL49UOF",
    "id": "41e48758-8b2b-474a-bd22-b0d6356025e2",
    "name": "DESKTOP-AL49UOF",
    "type": "filebeat",
    "version": "7.12.0"
  },
  "data_stream": {
    "dataset": "windows.powershell",
    "namespace": "default",
    "type": "logs"
  },
  "ecs": {
    "version": "1.8.0"
  },
  "elastic_agent": {
    "id": "ebe1c190-9026-11eb-97b7-670157a7fe8c",
    "version": "7.12.0"
  },
  "event": {
    "action": "Provider Lifecycle",
    "category": "process",
    "code": "600",
    "created": "2021-03-31T04:42:07.705Z",
    "dataset": "windows.powershell",
    "ingested": "2021-03-31T04:42:12.329Z",
    "kind": "signal",
    "module": "powershell",
    "provider": "PowerShell",
    "sequence": "5",
    "type": "info"
  },
  "host": {
    "architecture": "x86_64",
    "hostname": "DESKTOP-AL49UOF",
    "id": "8c87c48e-9f44-4504-abe1-2a4428adaa71",
    "ip": "2603:8001:bf00:f9c0::12c8,2603:8001:bf00:f9c0:e816:7a3a:71e2:3702,2603:8001:bf00:f9c0:7d5e:a714:f6a6:c52f,fe80::e816:7a3a:71e2:3702,192.168.1.132,2001:0:34f1:8072:3802:3b0e:3f57:fe7b,fe80::3802:3b0e:3f57:fe7b",
    "mac": "08:00:27:03:19:05,00:00:00:00:00:00:00:e0",
    "name": "DESKTOP-AL49UOF",
    "os": {
      "build": "15063.1387",
      "family": "windows",
      "kernel": "10.0.15063.1387 (WinBuild.160101.0800)",
      "name": "Windows 10 Pro",
      "platform": "windows",
      "type": "windows",
      "version": "10.0"
    }
  },
  "input": {
    "type": "winlog"
  },
  "log": {
    "level": "information"
  },
  "message": "Provider \"Environment\" is Started. \n\nDetails: \n\tProviderName=Environment\n\tNewProviderState=Started\n\n\tSequenceNumber=5\n\n\tHostName=ConsoleHost\n\tHostVersion=5.1.15063.1387\n\tHostId=c4373ed5-2fbc-405c-bcde-f5c370a32546\n\tHostApplication=powershell.exe -ExecutionPolicy Bypass -C Set-Service EventLog -StartupType Disabled\n\n\tEngineVersion=\n\tRunspaceId=\n\tPipelineId=\n\tCommandName=\n\tCommandType=\n\tScriptName=\n\tCommandPath=\n\tCommandLine=",
  "powershell": {
    "process": {
      "executable_version": "5.1.15063.1387"
    },
    "provider": {
      "name": "Environment",
      "new_state": "Started"
    }
  },
  "process": {
    "args": "powershell.exe,-ExecutionPolicy,Bypass,-C,Set-Service,EventLog,-StartupType,Disabled",
    "args_count": "8",
    "command_line": "powershell.exe -ExecutionPolicy Bypass -C Set-Service EventLog -StartupType Disabled",
    "entity_id": "c4373ed5-2fbc-405c-bcde-f5c370a32546",
    "title": "ConsoleHost"
  },
  "signal": {
    "_meta": {
      "version": "25"
    },
    "ancestors": "{\"id\":\"uyyWhngBHIiL0gIkyYbp\",\"type\":\"event\",\"index\":\".ds-logs-windows.powershell-default-2021.03.29-000001\",\"depth\":0}",
    "depth": "1",
    "original_event": {
      "action": "Provider Lifecycle",
      "category": "process",
      "code": "600",
      "created": "2021-03-31T04:42:07.705Z",
      "dataset": "windows.powershell",
      "ingested": "2021-03-31T04:42:12.329199354Z",
      "kind": "event",
      "module": "powershell",
      "provider": "PowerShell",
      "sequence": "5",
      "type": "info"
    },
    "original_time": "2021-03-31T04:42:06.367Z",
    "parent": {
      "depth": "0",
      "id": "uyyWhngBHIiL0gIkyYbp",
      "index": ".ds-logs-windows.powershell-default-2021.03.29-000001",
      "type": "event"
    },
    "parents": "{\"id\":\"uyyWhngBHIiL0gIkyYbp\",\"type\":\"event\",\"index\":\".ds-logs-windows.powershell-default-2021.03.29-000001\",\"depth\":0}",
    "rule": {
      "actions": "",
      "author": "ipninichuck",
      "created_at": "2021-03-31T04:38:02.741Z",
      "created_by": "ipninichuck",
      "description": "Windows Event Log has been set to not start-up on reboot. This is often done by an attacker to evade any future detections.",
      "enabled": "true",
      "exceptions_list": "",
      "false_positives": "",
      "filters": "",
      "from": "now-360s",
      "id": "e0a77120-91da-11eb-97b7-670157a7fe8c",
      "immutable": "false",
      "index": "logs-*,winlogbeat-*",
      "interval": "5m",
      "language": "eql",
      "license": "",
      "max_signals": "100",
      "meta": {
        "from": "1m",
        "kibana_siem_app_url": "https://denoir-stack.kb.us-west1.gcp.cloud.es.io:9243/s/siemplify/app/security"
      },
      "name": "Windows Event Log Disabled",
      "output_index": ".siem-signals-siemplify",
      "query": "process where event.module == \"powershell\" and\nprocess.args : ( \"powershell.exe\", \"Set-Service\", \"EventLog\", \"Disabled\")\n",
      "references": "",
      "risk_score": "73",
      "risk_score_mapping": "",
      "rule_id": "bebc93c7-c122-4223-9257-c0084b53aa24",
      "severity": "high",
      "severity_mapping": "",
      "tags": "Hosts ,Windows,threat detection ,Defense Evasion",
      "threat": "{\"framework\":\"MITRE ATT&CK\",\"tactic\":{\"id\":\"TA0005\",\"reference\":\"https://attack.mitre.org/tactics/TA0005\",\"name\":\"Defense Evasion\"},\"technique\":[{\"id\":\"T1562\",\"reference\":\"https://attack.mitre.org/techniques/T1562\",\"name\":\"Impair Defenses\",\"subtechnique\":[{\"id\":\"T1562.002\",\"reference\":\"https://attack.mitre.org/techniques/T1562/002\",\"name\":\"Disable Windows Event Logging\"}]}]}",
      "to": "now",
      "type": "eql",
      "updated_at": "2021-03-31T04:38:10.703Z",
      "updated_by": "ipninichuck",
      "version": "1"
    },
    "status": "open"
  },
  "winlog": {
    "api": "wineventlog",
    "channel": "Windows PowerShell",
    "computer_name": "DESKTOP-AL49UOF",
    "event_id": "600",
    "keywords": "Classic",
    "opcode": "Info",
    "provider_name": "PowerShell",
    "record_id": "41",
    "task": "Provider Lifecycle"
  }
}

</details>

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1071 

## Summary

Identifies attempts to disable EventLog and logman. This is often done by attackers in an attempt to evade detection on a system.

### CODE REVIEW

**Review by StevenDias33 — APPROVED** [review]

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+references = ["https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/logman"]
+risk_score = 21
+rule_id = "4de76544-f0e5-486a-8f84-eae0b6063cdc"
+severity = "low"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
```
Comment (Samirbous) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:26]: nice! the current query will return unrelated events (i.e. `cmd.exe /c delete file.txt`, you can adjust it as shown below:

```
process where event.type in ("start", "process_started") and
     ((process.name:"logman.exe" or process.pe.original_file_name == "Logman.exe") and 
           process.args : "EventLog-*" and process.args : ("stop", "delete")) or         
     ((process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in 
 ("pwsh.exe", "powershell.exe", "powershell_ise.exe")) and 
			process.args : "Set-Service" and process.args: "EventLog" and process.args : "Disabled")  or 
		  ((process.name:"auditpol.exe" or process.pe.original_file_name == "AUDITPOL.EXE") and process.args : "/success:disable")
```

Also there more methods to achieve the same that can be covered with same rule (e.g. auditpol):

https://svch0st.medium.com/event-log-tampering-part-1-disrupting-the-eventlog-service-8d4b7d67335c

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2021/05/06"
+maturity = "development"
+updated_date = "2021/05/06"
+
+[rule]
+author = ["Elastic","Ivan Ninichuck","Austin Songer"]
+description = """
+Identifies attempts to disable EventLog and logman. This is often done by attackers in an attempt to evade detection on
```
Comment (Samirbous) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:9]: if the rule purpose is to cover only logman, suggest to adjust the description to `Identifies attempts to disable EventLog via logman Windows utility. ....`

**Review by Samirbous — APPROVED** [review]

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+type = "eql"
+
+query = '''
+process where event.type in ("start", "process_started") and
+     ((process.name:"logman.exe" or process.pe.original_file_name == "Logman.exe") and 
+           process.args : "EventLog-*" and process.args : ("stop", "delete")) or         
+     ((process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in 
+ ("pwsh.exe", "powershell.exe", "powershell_ise.exe")) and 
+			process.args : "Set-Service" and process.args: "EventLog" and process.args : "Disabled")  or 
+		  ((process.name:"auditpol.exe" or process.pe.original_file_name == "AUDITPOL.EXE") and process.args : "/success:disable")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:33]:
```suggestion
process where event.type in ("start", "process_started") and

  ((process.name:"logman.exe" or process.pe.original_file_name == "Logman.exe") and 
      process.args : "EventLog-*" and process.args : ("stop", "delete")) or
               
  ((process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in 
      ("pwsh.exe", "powershell.exe", "powershell_ise.exe")) and 
	process.args : "Set-Service" and process.args: "EventLog" and process.args : "Disabled")  or 
    ((process.name:"auditpol.exe" or process.pe.original_file_name == "AUDITPOL.EXE") and process.args : "/success:disable")
```

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+[metadata]
+creation_date = "2021/05/06"
+maturity = "development"
+updated_date = "2021/05/06"
+
+
+[rule]
+author = ["Elastic","Ivan Ninichuck","Austin Songer"]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:8]:
```suggestion
author = ["Elastic", "Ivan Ninichuck", "Austin Songer"]
```

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+query = '''
+process where event.type in ("start", "process_started") and
+
+  ((process.name:"logman.exe" or process.pe.original_file_name == "Logman.exe") and 
+      process.args : "EventLog-*" and process.args : ("stop", "delete")) or
+               
+  ((process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in 
+      ("pwsh.exe", "powershell.exe", "powershell_ise.exe")) and 
+	process.args : "Set-Service" and process.args: "EventLog" and process.args : "Disabled")  or 
+    ((process.name:"auditpol.exe" or process.pe.original_file_name == "AUDITPOL.EXE") and process.args : "/success:disable")
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:35]:
```suggestion
process where event.type in ("start", "process_started") and

  ((process.name:"logman.exe" or process.pe.original_file_name == "Logman.exe") and
      process.args : "EventLog-*" and process.args : ("stop", "delete")) or

  ((process.name : ("pwsh.exe", "powershell.exe", "powershell_ise.exe") or process.pe.original_file_name in
      ("pwsh.exe", "powershell.exe", "powershell_ise.exe")) and
	process.args : "Set-Service" and process.args: "EventLog" and process.args : "Disabled")  or
	
  ((process.name:"auditpol.exe" or process.pe.original_file_name == "AUDITPOL.EXE") and process.args : "/success:disable")
```

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+[metadata]
+creation_date = "2021/05/06"
+maturity = "development"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:3]:
```suggestion
maturity = "production"
```

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+author = ["Elastic","Ivan Ninichuck","Austin Songer"]
+description = """
+Identifies attempts to disable EventLog via logman Windows utility. This is often done by attackers in an attempt to evade detection on
+a system.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Disable Windows Event and Security Logs"
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:17]:
```suggestion
name = "Disable Windows Event and Security Logs Using Built-in Tools"
```

File: rules/windows/defense_evasion_disabling_windows_logs.toml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+creation_date = "2021/05/06"
+maturity = "development"
+updated_date = "2021/05/06"
+
+
+[rule]
+author = ["Elastic","Ivan Ninichuck","Austin Songer"]
+description = """
+Identifies attempts to disable EventLog via logman Windows utility. This is often done by attackers in an attempt to evade detection on
+a system.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_disabling_windows_logs.toml:11]:
```suggestion
Identifies attempts to disable EventLog via the logman Windows utility, PowerShell, or auditpol. This is often done by 
attackers in an attempt to evade detection on a system.
```

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**